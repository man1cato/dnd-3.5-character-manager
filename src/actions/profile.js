import database from '../firebase/firebase'


//SET PROFILE IN STORE
export const setProfile = (id, profile) => ({
    type: 'SET_PROFILE',
    id,
    profile
});

//READ PROFILE FROM FIREBASE
export const startSetProfile = (id) => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid;
        return database.ref(`users/${uid}/profiles/${id}`).once('value').then((snapshot) => {
            const profile = snapshot.val();
            dispatch(setProfile(id, profile));
        });
    };
};

//CREATE PROFILE IN FIREBASE
export const startCreateProfile = (profile) => {    
    return (dispatch, getState) => {
        const uid = getState().auth.uid       
        return database.ref(`users/${uid}/profiles`).push(profile).then((ref) => {
            dispatch(setProfile(ref.key, profile))
        })
    }
}


//UPDATE PROFILE IN STORE
export const editProfile = (updates) => ({
    type: 'EDIT_PROFILE',
    updates
});

//UPDATE PROFILE IN FIREBASE
export const startEditProfile = (id, updates) => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid;
        return database.ref(`users/${uid}/profiles/${id}`).update(updates).then(() => {
            dispatch(editProfile(updates));
        });
    };
};

//DELETE PROFILE FROM STORE
export const deleteProfile = () => ({
    type: 'DELETE_PROFILE'
})

//LEVEL UP CHARACTER IN STORE
// export const levelUp = (character) => ({
//     type: 'LEVEL_UP',
//     character
// });
