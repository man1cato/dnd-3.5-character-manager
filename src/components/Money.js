import React from 'react';

const Money = ({money, onInputChange}) => (
    <div>
        <h3 className="row__money">Money</h3>
        <div className="row__money">
            <input
                type="text"
                name="money"
                id="pp"
                value={money.pp}
                onChange={onInputChange}
            />  
            <span>pp</span>
        </div>
        <div className="row__money">
            <input
                type="text"
                name="money"
                id="gp"
                value={money.gp}
                onChange={onInputChange}
            />  
            <span>gp</span>
        </div>
        <div className="row__money">
            <input
                type="text"
                name="money"
                id="sp"
                value={money.sp}
                onChange={onInputChange}
            />  
            <span>sp</span>
        </div>
        <div className="row__money">
            <input
                type="text"
                name="money"
                id="cp"
                value={money.cp}
                onChange={onInputChange}
            />  
            <span>cp</span>            
        </div>
        <div className="row__money">
            <h4>Total:</h4> <span>{money.total} gp</span> 
        </div>
    </div>    
)

export default Money