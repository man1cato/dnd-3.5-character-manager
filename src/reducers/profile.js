export default (state = {}, action) => {
    switch (action.type) {
        case 'SET_PROFILE':
            return {
                id: action.id,
                ...action.fields
            };
        case 'EDIT_PROFILE':
            return {
                ...state,
                ...action.updates
            };
        default:
            return state;
    }
};
