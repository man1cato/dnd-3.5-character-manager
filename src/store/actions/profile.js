import database from '../../firebase/firebase'
import { startGetProfiles } from './profiles'

//SET SELECTED PROFILE IN STORE
export const setProfile = (id, profile) => ({
    type: 'SET_PROFILE',
    id,
    profile
})

//READ PROFILE FROM FIREBASE
export const startSetProfile = (uid, id) => {
    return (dispatch) => {
        return database.ref(`users/${uid}/profiles/${id}`).once('value').then((snapshot) => {
            const profile = snapshot.val()
            dispatch(setProfile(id, profile))
        })
    }
}

//CREATE PROFILE IN FIREBASE
export const startCreateProfile = (profile) => {    
    return (dispatch, getState) => {
        const uid = getState().auth.uid       
        return database.ref(`users/${uid}/profiles`).push(profile).then((ref) => {
            dispatch(setProfile(ref.key, profile))
            dispatch(startGetProfiles(uid))
            localStorage.setItem('selectedCharacterId', ref.key)
        })
    }
}


//UPDATE PROFILE IN STORE
export const editProfile = (updates) => ({
    type: 'EDIT_PROFILE',
    updates
})

//UPDATE PROFILE IN FIREBASE
export const startEditProfile = (id, updates) => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid
        return database.ref(`users/${uid}/profiles/${id}`).update(updates).then(() => {
            dispatch(editProfile(updates))
        })
    }
}

//REMOVE PROFILE FROM STORE
export const removeProfile = () => ({
    type: 'REMOVE_PROFILE'
})

//REMOVE PROFILE FROM FIREBASE
// export const startRemoveProfile = () => {
//     return (dispatch) => {
//         return database.ref(`users/${uid}/profiles/${id}`).remove().then(() => {
//             dispatch(removeProfile())
//         })
//     }
// }

//LEVEL UP CHARACTER IN STORE
// export const levelUp = (character) => ({
//     type: 'LEVEL_UP',
//     character
// });
