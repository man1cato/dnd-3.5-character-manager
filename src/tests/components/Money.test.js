import React from 'react';
import { shallow } from 'enzyme';

import Money from '../../components/Money';
import profile from '../fixtures/profile';

const money = profile.fields.money;

test('should render Money with profile data', () => {
    const wrapper = shallow(<Money money={money} />);
    expect(wrapper).toMatchSnapshot();
})
