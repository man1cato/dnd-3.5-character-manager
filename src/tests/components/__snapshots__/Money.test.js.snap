// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Money with profile data 1`] = `
<div>
  <h3
    className="row__money"
  >
    Money
  </h3>
  <div
    className="row__money"
  >
    <input
      id="pp"
      name="money"
      type="text"
      value={0}
    />
    <span>
      pp
    </span>
  </div>
  <div
    className="row__money"
  >
    <input
      id="gp"
      name="money"
      type="text"
      value={1200}
    />
    <span>
      gp
    </span>
  </div>
  <div
    className="row__money"
  >
    <input
      id="sp"
      name="money"
      type="text"
      value={350}
    />
    <span>
      sp
    </span>
  </div>
  <div
    className="row__money"
  >
    <input
      id="cp"
      name="money"
      type="text"
      value={80}
    />
    <span>
      cp
    </span>
  </div>
  <div
    className="row__money"
  >
    <h4>
      Total:
    </h4>
     
    <span>
       gp
    </span>
  </div>
</div>
`;
